spio_hash.h not installed together with pio.h

In pio.h, spio_hash.h is explicitly included:
`#include "spio_hash.h" /* Hash table implementation for ADIOS read */`

However, this file is not installed together with pio.h under the include directory.

Steps to reproduce this issue on ANL Ubuntu 20 workstations:
```
module load cmake/3.20.5-zyz2eld
module load gcc/11.1.0-qsjmpcg
export PATH=/nfs/gce/projects/climate/software/linux-ubuntu20.04-x86_64/mpich/4.0/gcc-11.1.0/bin:$PATH

git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio

mkdir build
cd build

ADIOS2_DIR=/nfs/gce/projects/climate/software/linux-ubuntu20.04-x86_64/adios2/2.9.1/mpich-4.0/gcc-11.1.0 \
CC=mpicc CXX=mpicxx FC=mpifort cmake -Wno-dev \
-DCMAKE_INSTALL_PREFIX=$HOME/spio \
-DWITH_ADIOS2=ON \
-DWITH_NETCDF=OFF \
-DPnetCDF_PATH=/nfs/gce/projects/climate/software/linux-ubuntu20.04-x86_64/pnetcdf/1.12.2/mpich-4.0/gcc-11.1.0 \
-DPIO_USE_MALLOC=ON \
..

make

make install

ls $HOME/spio/include/*.h
```

This issue was initially mentioned in #554 by a user of ESMF and SCORPIO:

> I tried to compile scorpio with esmf. I included scorpio include path and adios2 include path. But I still got this error:
> In file included from /glade/.../esmf/src/Infrastructure/IO/src/ESMCI_IO_Handler.C:36:
> In file included from /glade/.../esmf/src/Infrastructure/IO/src/../include/ESMCI_PIO_Handler.h:45:
> /glade/.../scorpio/install/include/pio.h:71:10: fatal error: In file included from /glade/.../esmf/src/Infrastructure/IO/src/ESMCI_PIO_Handler.C:'spio_hash.h' file not found27:

--- a/src/clib/pio.h
+++ b/src/clib/pio.h
@@ -68,8 +68,6 @@ extern "C" {
 #include <limits.h>
 #include <adios2_c.h>
 
-#include "spio_hash.h" /* Hash table implementation for ADIOS read */
-
 #define MAX_ADIOS_BUFFER_COUNT (PIO_MAX_CACHED_STEPS_FOR_ADIOS + 16) /* Maximum buffer size for aggregating decomp_id, frame_id, and fillval_id values */
 #define BLOCK_MAX_BUFFER ((unsigned long)INT_MAX) /* 2GB limit of MPI_Gatherv */
 /* adios end step is called if the number of blocks written out exceeds BLOCK_COUNT_THRESHOLD */
@@ -334,6 +332,9 @@ typedef PIO_OFFSET_C_TYPENAME PIO_Offset;
 /** Some fwd declarations to avoid including internal headers */
 typedef struct mtimer_info *mtimer_t;
 #endif
+
+/* Forward decl of hash map used for ADIOS reads */
+struct spio_hmap;
 
 /**
  * Variable description structure.
@@ -1075,9 +1076,9 @@ typedef struct file_desc_t
     int current_frame;
 
     /* Some caches (hash tables) for ADIOS read */
-    spio_hash_t *cache_data_blocks;
-    spio_hash_t *cache_block_sizes;
-    spio_hash_t *cache_darray_info;
+    struct spio_hmap *cache_data_blocks;
+    struct spio_hmap *cache_block_sizes;
+    struct spio_hmap *cache_darray_info;
 
     char io_name_reader[PIO_MAX_NAME + 1]; /* Name of io object, for ADIOS read */
 #endif /* _ADIOS2 */

--- a/src/clib/pio_darray.c
+++ b/src/clib/pio_darray.c
@@ -17,6 +17,7 @@
 #include "pio_sdecomps_regex.h"
 #include "spio_io_summary.h"
 #include "spio_file_mvcache.h"
+#include "spio_hash.h"
 
 /* uint64_t definition */
 #ifdef _ADIOS2

--- a/src/clib/pio_getput_int.c
+++ b/src/clib/pio_getput_int.c
@@ -13,6 +13,7 @@
 #include <pio.h>
 #include <pio_internal.h>
 #include "spio_io_summary.h"
+#include "spio_hash.h"
 
 /**
  * Write a netCDF attribute of any type, converting to any type.

--- a/src/clib/pio_lists.c
+++ b/src/clib/pio_lists.c
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include "spio_file_mvcache.h"
 #include "spio_io_summary.h"
+#include "spio_hash.h"
 
 static io_desc_t *pio_iodesc_list = NULL;
 static io_desc_t *current_iodesc = NULL;

--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -21,6 +21,7 @@
 #endif
 #include "spio_io_summary.h"
 #include "spio_file_mvcache.h"
+#include "spio_hash.h"
 
 #define VERSNO 2001
 

